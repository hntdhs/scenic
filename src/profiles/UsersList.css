.user-link {
    font-weight: bold;
    font-family: fantasy;
    font-size: x-large;
    width: 33%;
    margin: .5em;
    text-align: center;
}

.user-link a {
    color: green;
    font-family: fantasy;
}

.user-list-explain {
    text-align: center;
    font-size: large;
    margin-bottom: 1em;
}

#users-list .float-child {
    float: left;
}