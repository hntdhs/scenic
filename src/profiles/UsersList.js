import React, { useState, useEffect } from "react";
import BywayApi from "../api/api";
import { Link } from "react-router-dom";
import "./UsersList.css"


function UsersList() {

    const [users, setUsers] = useState('');

    useEffect(() => {
        async function getUsers() {
            const u = await BywayApi.getAllUsers();
            setUsers(u.users);
        }
        getUsers();
    }, []);

    return (
        <div className="float-container" id="users-list">
            <div class="container text-center" id="app-name">
                <Link class="link-offset-2 link-underline link-underline-opacity-0 homepage-link" to="/"><h1>NATIONAL SCENIC BYWAYS</h1></Link>
            </div>
            <div className="user-list-explain">A list of all users and a link to their profiles</div>
            <div className="float-child w-50">
                {users.length
                ? users.map((u, i) => (
                            <div className="user-link">

                            <a key={i} href={`/profile/${u.username}`} >{u.username}</a>
                            </div>
                        ))
                
                : (
                    <h4>Sorry, no results found.</h4>
                )}
            </div>
            <div className="float-child w-50">
                <div className="gallery">
                    <figure class="gallery__item gallery__item--1">
                        <img src="" class="gallery__img" alt="" />
                    </figure>
                    <figure class="gallery__item gallery__item--2">
                        <img src="" class="gallery__img" alt="" />
                    </figure>
                    <figure class="gallery__item gallery__item--3">
                        <img src="" class="gallery__img" alt="" />
                    </figure>
                    <figure class="gallery__item gallery__item--4">
                        <img src="" class="gallery__img" alt="" />
                    </figure>
                    <figure class="gallery__item gallery__item--5">
                        <img src="" class="gallery__img" alt="" />
                    </figure>
                    <figure class="gallery__item gallery__item--6">
                        <img src="" class="gallery__img" alt="" />
                    </figure>        
                </div>
            </div>
        </div>
        
    )
}

export default UsersList;