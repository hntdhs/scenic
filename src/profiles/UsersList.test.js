import React from "react";
import { render } from "@testing-library/react";
import UsersList from "./UsersList";

it("matches snapshot", function () {
    const { asFragment } = render(<UsersList />);
    expect(asFragment()).toMatchSnapshot();
  });