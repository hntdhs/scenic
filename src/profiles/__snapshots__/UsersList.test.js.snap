// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container text-center"
      id="app-name"
    >
      <h1>
        NATIONAL SCENIC BYWAYS
      </h1>
    </div>
    <h4>
      Sorry, no results found.
    </h4>
    <div
      class="gallery"
    >
      <figure
        class="gallery__item gallery__item--1"
      >
        <img
          alt=""
          class="gallery__img"
          src=""
        />
      </figure>
      <figure
        class="gallery__item gallery__item--2"
      >
        <img
          alt=""
          class="gallery__img"
          src=""
        />
      </figure>
      <figure
        class="gallery__item gallery__item--3"
      >
        <img
          alt=""
          class="gallery__img"
          src=""
        />
      </figure>
      <figure
        class="gallery__item gallery__item--4"
      >
        <img
          alt=""
          class="gallery__img"
          src=""
        />
      </figure>
      <figure
        class="gallery__item gallery__item--5"
      >
        <img
          alt=""
          class="gallery__img"
          src=""
        />
      </figure>
      <figure
        class="gallery__item gallery__item--6"
      >
        <img
          alt=""
          class="gallery__img"
          src=""
        />
      </figure>
    </div>
  </div>
</DocumentFragment>
`;
